src/services/slides_loader.py

```
@@ -0,0 +1,105 @@
+
+                functions[func_name] = {
+                    "function": func,
+                    "docstring": inspect.getdoc(func) or "",
+                    "signature": str(inspect.signature(func)),
+                    "args_info": self._parse_function_args(func),
+                }
+
+            return functions
+
+        except ImportError as e:
+            raise ImportError(f"Cannot load template '{template_id}': {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)